runtime/kusama/src/weights/pallet_staking.rs

```diff
@@ -42,181 +44,368 @@ use sp_std::marker::PhantomData;
+			.saturating_add((16_204_000 as Weight).saturating_mul(s as Weight))
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
-	fn payout_stakers_dead_controller(n: u32) -> Weight {
-		(108_055_000 as Weight)
-			// Standard Error: 15_000
-			.saturating_add((49_008_000 as Weight).saturating_mul(n as Weight))
+	// Storage: Staking CurrentEra (r:1 w:0)
+	// Storage: Staking HistoryDepth (r:1 w:0)
+	// Storage: Staking ErasValidatorReward (r:1 w:0)
+	// Storage: Staking Bonded (r:2 w:0)
+	// Storage: Staking Ledger (r:1 w:1)
+	// Storage: Staking ErasStakersClipped (r:1 w:0)
+	// Storage: Staking ErasRewardPoints (r:1 w:0)
+	// Storage: Staking ErasValidatorPrefs (r:1 w:0)
+	// Storage: Staking Payee (r:2 w:0)
+	// Storage: System Account (r:2 w:2)
+	fn payout_stakers_dead_controller(n: u32, ) -> Weight {
+		(113_536_000 as Weight)
+			// Standard Error: 16_000
+			.saturating_add((45_494_000 as Weight).saturating_mul(n as Weight))
 			.saturating_add(T::DbWeight::get().reads(10 as Weight))
 			.saturating_add(T::DbWeight::get().reads((3 as Weight).saturating_mul(n as Weight)))
 			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(n as Weight)))
 	}
-	fn payout_stakers_alive_staked(n: u32) -> Weight {
-		(122_624_000 as Weight)
-			// Standard Error: 22_000
-			.saturating_add((60_815_000 as Weight).saturating_mul(n as Weight))
+	// Storage: Staking CurrentEra (r:1 w:0)
+	// Storage: Staking HistoryDepth (r:1 w:0)
+	// Storage: Staking ErasValidatorReward (r:1 w:0)
+	// Storage: Staking Bonded (r:2 w:0)
+	// Storage: Staking Ledger (r:2 w:2)
+	// Storage: Staking ErasStakersClipped (r:1 w:0)
+	// Storage: Staking ErasRewardPoints (r:1 w:0)
+	// Storage: Staking ErasValidatorPrefs (r:1 w:0)
+	// Storage: Staking Payee (r:2 w:0)
+	// Storage: System Account (r:2 w:2)
+	// Storage: Balances Locks (r:2 w:2)
+	fn payout_stakers_alive_staked(n: u32, ) -> Weight {
+		(134_960_000 as Weight)
+			// Standard Error: 25_000
+			.saturating_add((58_130_000 as Weight).saturating_mul(n as Weight))
 			.saturating_add(T::DbWeight::get().reads(11 as Weight))
 			.saturating_add(T::DbWeight::get().reads((5 as Weight).saturating_mul(n as Weight)))
 			.saturating_add(T::DbWeight::get().writes(3 as Weight))
 			.saturating_add(T::DbWeight::get().writes((3 as Weight).saturating_mul(n as Weight)))
 	}
-	fn rebond(l: u32) -> Weight {
-		(46_725_000 as Weight)
-			// Standard Error: 1_000
-			.saturating_add((61_000 as Weight).saturating_mul(l as Weight))
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(3 as Weight))
+	// Storage: Staking Ledger (r:1 w:1)
+	// Storage: Balances Locks (r:1 w:1)
+	// Storage: System Account (r:1 w:1)
+	// Storage: BagsList ListNodes (r:3 w:3)
+	// Storage: Staking Bonded (r:1 w:0)
+	// Storage: BagsList ListBags (r:2 w:2)
```

Comment:
would be nice to make a follow up issue to consider refund for these operations. All of this is the worst case now (which is totally intentional)

Reply:
https://github.com/paritytech/substrate/issues/9826